source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+
+Like usual, a few initializers are added, there are a few changes in config/environments/, an app/models/application_record.rb is added and the models now inherit from ApplicationRecord. We can influence these changes.
+
+Within the new initializers, `config/initializers/active_record_belongs_to_required_by_default.rb` is added, which makes it so that the belongs_to cannot be nil. In our case, we need some to be nil, so we have to add the option `optional: true` to the relations. For example, `belongs_to :category, optional: true`.
+
+Also, config/initializers/callback_terminator.rb is added, which makes it so that the callbacks do not halt more when a false is returned. The way to halt now is with `throw(:abort)`. In our application, this was irrelevant.
+
+After incorporating all the changes, we return to running the tests and testing the application by hand, and everything functions wonderfully.
+
+
+## Conclusion
+
+In general, it’s not as easy as it might seem to update to a Rails beta. We have to be ready to deal with some external problems, debug third-party code, etc.
+
+Not all the gems we use are ready to work with Rails 5.0.0.beta1. In the case that they don’t work, we could check if the development versions of those gems function or if there’s a branch of Rails 5, or something of that nature. Otherwise, we must resort to another solution or adapt the gem to Rails 5.
+
+Considering all these difficulties, we may ask ourselves why we should update our application to a beta version of Rails and not wait for the final version (or at least the release candidate). There are a few good reasons for this: It’s a good way to anticipate the problems that our new application can have with the new version of Rails, and besides being able to report possible problems and thus assure ourselves that the new version of Rails is going to work correctly with our application, we can also verify the state of the gems that our project utilizes. It’s also an excellent way to contribute to the development of Rails, given that the testing of the beta version depends on the community. And, it serves as a learning experience and contributes to Rails – even if it’s just reporting the problems that arise during the migration.
+
+It’s quite probable that if you update your applications, you will come across some issue that is not covered in this article. My recommendation is to follow a process similar to the one we followed in this article and to be ready to fix the problems that come up in Rails itself or in one of the gems that we use.
+
+The beta2 or rc1 is going to be released on the 1st of February and is already going to resolve all of these problems. However, as mentioned throughout the article, if the released or “stable” version doesn’t work for us, we should try with the master directly.
```

Comment:
`we should try with the master directly` => `we should try with master directly`